lnbits/core/crud.py

```diff
@@ -602,6 +602,7 @@ async def get_standalone_payment(
         SELECT *
         FROM apipayments
         WHERE {clause}
+        ORDER BY amount
```

Comment:
I don't understand the logic why sorting by amount is desired and not by some other field (e.g. time comes to my mind).

Reply:
i think its for internal payments, the one sending is negative the one receiving positive

Reply:
The original issue was that the `extra` field was sometimes not correctly fetched for get payment (`/api/v1/payments/${paymenHash}`). The `extra` field contains (among others the webhook status).

The `extra` field is updated like so on `core` and `withdraw` extension:
```py
            await update_payment_extra(
                payment_hash=payment_hash,
                extra={
                    "wh_success": r.is_success,
                    "wh_message": r.reason_phrase,
                    "wh_response": r.text,
                },
                outgoing=True,
            )
```

Having `outgoing=True` makes the update on the outgoing (negative) payment.
![telegram-cloud-photo-size-4-6006064172485427192-y](https://github.com/lnbits/lnbits/assets/2951406/60aac06c-e365-454d-9e75-bbc601653caa)

However, now I have just noticed that [the call from lnurlp extension](https://github.com/lnbits/lnurlp/blob/main/tasks.py#L95-L106) does not specify ` outgoing=True`.
This is indeed inconsistent (and fragile).

Reply:
> However, now I have just noticed that [the call from lnurlp extension](https://github.com/lnbits/lnurlp/blob/main/tasks.py#L95-L106) does not specify  outgoing=True.
This is indeed inconsistent (and fragile).

- nvm, `lnurlp` is not an outgoing payment. Its fine as it is.